app/src/main/java/com/vultisig/wallet/ui/screens/vault_settings/components/ConfirmDeleteScreen.kt

```diff
@@ -0,0 +1,203 @@
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.ModalBottomSheet
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.material3.rememberModalBottomSheetState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.vultisig.wallet.R
+import com.vultisig.wallet.ui.components.MultiColorButton
+import com.vultisig.wallet.ui.components.UiSpacer
+import com.vultisig.wallet.ui.theme.Theme
+import com.vultisig.wallet.ui.theme.dimens
+import kotlinx.coroutines.launch
+
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+internal fun ConfirmDeleteScreen(
+    cautions: List<Int>,
+    checkedCautionIndexes: List<Int>,
+    isDeleteButtonActive: Boolean,
+    onDismissClick: () -> Unit,
+    onItemCheckChangeClick: (Int, Boolean) -> Unit,
+    onConfirmClick: () -> Unit
+) {
+    val textColor = MaterialTheme.colorScheme.onBackground
+    val appColor = Theme.colors
+    val dimens = MaterialTheme.dimens
+    val coroutineScope = rememberCoroutineScope()
+    val modalBottomSheetState = rememberModalBottomSheetState(
+        skipPartiallyExpanded = true
+    )
+
+    ModalBottomSheet(
+        containerColor = Color.Transparent,
+        sheetState = modalBottomSheetState,
+        onDismissRequest = {
+            coroutineScope.launch {
+                modalBottomSheetState.hide()
+                onDismissClick()
+            }
+        }) {
+        Scaffold(
+            topBar = {
+                Row(
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .padding(all = 16.dp),
+                    horizontalArrangement = Arrangement.SpaceBetween,
+                    verticalAlignment = Alignment.CenterVertically
+                ) {
+
+                    Image(
+                        painter = painterResource(id = R.drawable.caret_left),
+                        contentDescription = null,
+                        modifier = Modifier
+                            .size(24.dp)
+                            .clickable(onClick = {
+                                coroutineScope.launch {
+                                    modalBottomSheetState.hide()
+                                    onDismissClick()
+                                }
+                            }),
+                    )
+
+                    UiSpacer(weight = 1f)
+
+                    Text(
+                        text = "Delete Vault",
```

Comment:
Let's keep all strings in strings.xml. So we don't have to do #131 over again.

Reply:
Done